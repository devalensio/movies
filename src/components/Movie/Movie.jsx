import React, { useState, useRef } from 'react';

import { useOutsideClick } from '../../utils/hooks/useOutsideClick';

import styles from './Movie.module.scss';

const Movie = ({ movie, handleOnClick }) => {
  const [isPosterOpen, setPosterOpen] = useState(false);

  const imageUrl = movie.Poster === 'N/A' ? '/images/not-found.png' : movie.Poster;
  return (
    <div className={styles.movie__wrapper}>
      <div className={styles['movie__wrapper--header']}>
        <img
          src={imageUrl}
          alt="Movie Poster"
          height="250px"
          width="100%"
          onClick={() => setPosterOpen(true)}
        />
        <h1 onClick={handleOnClick}>{movie.Title}</h1>
      </div>
      <p>Type: {movie.Type}</p>
      <p>Release Date: {movie.Year}</p>
      {
        isPosterOpen && <ShowPosterView moviePoster={imageUrl} closeModal={() => setPosterOpen(false)} />
      }
    </div>
  );
};

export default Movie;

const ShowPosterView = ({ moviePoster, closeModal }) => {
  const ref = useRef();

  useOutsideClick(ref, () => {
    closeModal();
  });

  return (
    <div className={styles.movie__modal}>
      <div className={styles['movie__modal--content']} ref={ref}>
        <div className={styles.content__block}>
          <span onClick={closeModal}>âœ–</span>
          </div>
        <img src={moviePoster} alt="Movie Poster" />
      </div>
    </div>
  );
};
