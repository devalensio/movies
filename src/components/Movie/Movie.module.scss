@import "../../styles/tokens/all";

.movie__wrapper {
  margin-bottom: $space--md;
  border-bottom: 1px solid $color-lp-grey;
  &--header {
    display: flex;
    flex-direction: column;
    img {
      object-fit: cover;
      cursor: pointer;
    }
    h1 {
      text-align: center;
      cursor: pointer;
      &:hover {
        text-decoration: underline;
      }
    }
  }
}

.movie__modal {
  position: fixed;
  z-index: 1;
  padding-top: 100px;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  overflow: auto;
  background-color: rgba(0, 0, 0, 0.4);
  &--content {
    background-color: $color--white;
    margin: auto;
    padding: $space--md;
    border: 1px solid $color-grey-70;
    border-radius: $space--2xs;
    width: max-content;
    .content__block {
      display: flex;
      justify-content: flex-end;
      span {
        margin-bottom: $space--sm;
        text-align: right;
        cursor: pointer;
      }
    }
  }
}
