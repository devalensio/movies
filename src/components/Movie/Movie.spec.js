import React from 'react';

import renderer from 'react-test-renderer';

import Movie from './Movie';

describe('Movie', () => {
  it('renders Movie successfully', () => {
    const component = renderer.create(<Movie movie={{ Poster: 'N/A' }} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
