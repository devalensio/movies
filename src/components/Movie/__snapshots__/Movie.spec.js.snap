// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie renders Movie successfully 1`] = `
<div
  className="movie__wrapper"
>
  <div
    className="movie__wrapper--header"
  >
    <img
      alt="Movie Poster"
      height="250px"
      onClick={[Function]}
      src="/images/not-found.png"
      width="100%"
    />
    <h1 />
  </div>
  <p>
    Type: 
  </p>
  <p>
    Release Date: 
  </p>
</div>
`;
